File: Problem_of_the_day_Algorithms/ShortDistChar.py
class Solution:
    def shortestToChar(self, s: str, c: str) -> List[int]:
        #create 2 lists position and output(finding min dist of each char in lis with ch
        pos = []
        output= []
        n = len(s)
        for i in range(n):
            if s[i] == c:
                pos.append(i)
        for i in range(n):
            min_dist = n
            for j in pos:
                min_dist = min(min_dist, abs(i-j))
            output.append(min_dist)
        return output
                               

File: Problem_for_the_Day/ShortDistChar.cpp
class Solution {
public:
    vector<int> shortestToChar(string s, char ch) {
        vector<int> position;
        vector<int> output;
        int c=0;
        int i=0;
        int index=-1;
        for(int i=0;i<s.size();i++){
            if(s[i]==ch){
                index=i;
                position.push_back(index);
            } 
        }
        for(int i=0;i<s.size();i++){
            int minimum_distance=INT_MAX;
            for(int j=0;j<position.size();j++){
                minimum_distance=min(minimum_distance,abs(i-position[j]));
            }
            output.push_back(minimum_distance);
        }
       //osition.push_back(index);
        // while(i<s.size()){
        //     while(s[i]!=ch){
        //         c++;
        //         i++;
        //         cout<<c<<"\n";
        //     }
        //     i=1;
        //     result.push_back(c);
        // }
        // for(int i=0;i<result.size();i++){
        //    cout<<result[i]<<" "; 
        // }
        return output;
    }
};
